### ISSUE TITLE
Libtiff isn't included with Leptonica 

### ISSUE THREAD

**Issue by maximumspatium** [bytedeco/javacpp-presets#214]
While fooling around with the Leptonica preset I've noticed that the native code of the Leptonica's shared object doesn't actually include Libtiff. That's not how it's supposed to work.

Leptonica's autoconfigure script checks for presence of several I/O libraries (libjpeg, libtiff, libgif etc.) and enables their linking automatically. The problem is that the check for Libtiff actually fails:

checking for cos in -lm... yes
checking for deflate in -lz... yes
checking for png_read_png in -lpng... yes
checking for jpeg_read_scanlines in -ljpeg... yes
checking for DGifOpenFileHandle in -lgif... yes
**checking for TIFFOpen in -ltiff... no**

Why does it happen? After some investigation I found out that the test code in the configure.ac script misses the "-ljbig" dependency in the gcc invocation. Below a short excerpt from config.log:

```
configure:12538: checking for TIFFOpen in -ltiff
configure:12564: gcc -m32 -fPIC -o conftest -pthread -I/home/maxpol/Dokumente/javacpp-presets/leptonica/cppbuild/linux-x86/include/  -L/home/maxpol/Dokumente/javacpp-presets/leptonica/cppbuild/linux-x86/lib/ conftest.c -ltiff -lm -lz -ljpeg
    >&5
...libtiff.a(tif_jbig.o): In function `JBIGEncode':
...tif_jbig.c:164: undefined reference to `jbg_enc_init'
collect2: error: ld returned 1 exit status
```
Therefore I've patched the corresponding line in the configure.ac of Leptonica to include "-ljbig". Although it made the test for Libtiff work as expected, Leptonica's compilation still fails at the final linking because the global -ljbig dependency is missing as well.

This bug has something to do with the fact that the Leptonica preset links all required I/O libs statically. Disabling static linking by removing the "--disable-shared" option makes everything work as expected. Therefore I suppose that the JBIG dependency will be somehow silently included when linking dynamically. That may be the reason why noone has discovered that bug until now.

I'm not an Autotools guru so I don't know how to fix this properly. It looks to me like a Leptonica bug that should be reported to the upstream project ASAP. Although it's not a failure of the Javacpp-presets project, the Leptonica/Tesseract preset will remain unusable for projects relying on TIFF support (like one of my current projects), until a fix will be provided.

Best regards
Max

P.S.: I'm using the following command to build Leptonica on Linux-x86:
bash cppbuild.sh install leptonica

**Comment by saudet** [comment]
Works fine for me with both linux-x86 and linux-x86_64:
```
checking for cos in -lm... yes
checking for deflate in -lz... yes
checking for png_read_png in -lpng... yes
checking for jpeg_read_scanlines in -ljpeg... yes
checking for DGifOpenFileHandle in -lgif... yes
checking for TIFFOpen in -ltiff... yes
checking for WebPGetInfo in -lwebp... yes
checking for opj_create_decompress in -lopenjp2... no
```
Please make sure to follow the instructions here:
https://github.com/bytedeco/javacpp-presets/wiki/Build-Environments#linux-x86-and-x86_64

**Comment by maximumspatium** [comment]
Hello Samuel,

thank you for pointing me to the updated build documentation. Unfortunately, it doesn't work for me because Docker.io doesn't support 32bit Linux hosts. I'll therefore try to get around this issue by applying local patches in order to be able to compile binaries myself. Anyway, the binaries distributed with Javacpp-presets should work on my machine just out of box.

**Comment by saudet** [comment]
Anyway, I've released version 1.2 with updated binaries, so just use those. They will work.
